torchgeo/datasets/spacenet.py

```diff
@@ -191,49 +199,287 @@ def __getitem__(self, index: int) -> Dict[str, Tensor]:
 
         return sample
 
-    def _check_integrity(self) -> bool:
+    def _check_integrity(self) -> List[str]:
         """Checks the integrity of the dataset structure.
 
         Returns:
-            True if the dataset directories are found, else False
+            List of collections be downloaded
         """
-        stacpath = os.path.join(self.root, self.foldername, "collection.json")
-
-        if os.path.exists(stacpath):
-            return True
-
-        # If dataset folder does not exist, check for uncorrupted archive
-        archive_path = os.path.join(self.root, self.foldername + ".tar.gz")
-        if not os.path.exists(archive_path):
-            return False
-        print("Archive found")
-        if self.checksum and not check_integrity(archive_path, self.md5):
-            print("Dataset corrupted")
-            return False
-        print("Extracting...")
-        extract_archive(archive_path)
-        return True
```

Comment:
These lines aren't covered by unit tests

Reply:
I think this is from the old version where datasets were not organised as collections. N/A